halo2_proofs/src/poly/multiopen/shplonk/prover.rs

```diff
@@ -208,8 +211,16 @@ where
         assert_eq!(must_be_zero, C::Scalar::zero());
     }
 
+    let mut h_x = div_by_vanishing(l_x, &[*u]);
+
+    // normalize coefficients by the coefficient of the first polynomial
+    let z_0_inv = z[0].invert().unwrap();
```

Comment:
Perhaps with a `diff` suffix?

```suggestion
    let z_0_diff_inv = z_diffs[0].invert().unwrap();
```